common.js

```diff
@@ -83,6 +83,11 @@ function setopts (self, pattern, options) {
   self.statCache = options.statCache || Object.create(null)
   self.symlinks = options.symlinks || Object.create(null)
 
+  // avoiding empty strings
+  if (!!options.pathsep) {
+    pattern = pattern.split(options.pathsep).join("/")
```

Comment:
While using String.prototype.replace with a plain RegExp with the `g` flag (String.prototype.replaceAll isn't supported on node v12 and v14) is significantly faster than splitting and joining strings, creating a RegExp from dynamic content poses both security risks and can throw incorrect errors when a single backslash is inputted. 
Escaping potentially dangerous characters will put a large load on one point security-wise, and is less performant than simply splitting and joining strings.